src/AryUtil.chpl

```diff
@@ -248,7 +248,9 @@ module AryUtil
 
     inline proc getBitWidth(a: [?aD] real): (int, bool) {
       const bitWidth = numBits(real);
-      const negs = signbit(min reduce a);
+      const minVal = min reduce a;
+      // nan is returned for a `min reduce` even if negatives are present in that case do an `| reduce` over all signbits
+      const negs = if isnan(minVal) then | reduce signbit(a) else signbit(minVal);
       return (bitWidth, negs);
```

Comment:
This doesn't use the minVal except to check the sign bit, so probably don't need to bother computing it and can instead just do the `const negs = | reduce signbit(a);`

Having the extra reduce today shouldn't have any noticeable performance impact. It's an extra read of the array, but that's really fast compared to the sorting itself.

Reply:
Awesome! I was thinking the same thing but wanted to make sure!

Reply:
And just seeing you mentioned that in the OP, but clearly I didn't read anything before my name :)

`signbit` is  trivially fast to call (shouldn't be any more expensive than the comparison used to check for `min`) so I'd just use that instead of trying to limit when you call signbit.